convert2rhel/pkgmanager/handlers/yum/__init__.py

```diff
@@ -345,23 +331,58 @@ def _process_transaction(self, validate_transaction):
         else:
             loggerinst.info("System packages replaced successfully.")
 
-    @utils.run_as_child_process
     def run_transaction(self, validate_transaction=False):
         """Run the yum transaction.
 
         Perform the transaction. If the `validate_transaction` parameter set to
         true, it means the transaction will not be executed, but rather verify
         everything and do an early return.
 
+        :param validate_transaction: Determines if the transaction needs to be
+            validated or not.
+        :type validate_transaction: bool
+        :raises CriticalError: If we can't resolve the transaction dependencies.
+        """
+        resolve_deps_finished = False
+        # Do not allow this to loop until eternity.
+        attempts = 0
+        try:
+            while attempts <= MAX_NUM_OF_ATTEMPTS_TO_RESOLVE_DEPS:
+                self._set_up_base()
```

Comment:
Why is the setup moved here from the `perform_operations`?

Reply:
Answered in slack.